### ISSUE TITLE
ModelCheckpoint is conditioned on batch level metrics 

### ISSUE THREAD

**Issue by egafni** [Lightning-AI/lightning#12954]
## 🐛 Bug

The most common use case of ModelCheckPoint is to save using a validation metric such as "val/loss".  However, in its current behavior, it uses the current state of trainer.callback_metrics which contains the last result of the last metric update from a call to self.log().  

### Expected behavior

The desired and expected behavior would be to use the loss or accuracy aggregated over the entire validation set.

cc @carmocca @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy

**Comment by rohitgr7** [comment]
> The desired and expected behavior would be to use the loss or accuracy aggregated over the entire validation set.

it does use the aggregated metrics, if you are logging it using `self.log(..., on_step=False, on_epoch=True)` which is also done by default if you are logging it inside validation hooks.

ref: https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html#id3
